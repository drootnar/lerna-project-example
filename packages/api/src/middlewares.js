import bodyParser from 'body-parser'

export function applyMiddlewares(app) {
  app.use(bodyParser.json())
  app.use(bodyParser.urlencoded({ extended: false }))
}

export function applyErrorMiddlewares(app) {}
