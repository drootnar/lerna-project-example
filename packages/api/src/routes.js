import { Router } from 'express'

const routes = Router()

routes.get('', (req, res, next) => {
  res.send({ status: 'ok' })
})
export default routes
