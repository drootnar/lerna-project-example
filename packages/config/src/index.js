export default {
  logLevel: process.env.LOG_LEVEL || 'NONE',
  api: {
    port: 5500
  }
}
