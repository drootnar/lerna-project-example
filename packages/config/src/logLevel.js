export default process.env.LOG_LEVEL || 'debug'
