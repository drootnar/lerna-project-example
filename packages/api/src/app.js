import express from 'express'
import { applyMiddlewares, applyErrorMiddlewares } from './middlewares'

export function createApp(router) {
  const app = express()
  applyMiddlewares(app)
  app.use('/', router)
  applyErrorMiddlewares(app)
  return app
}
