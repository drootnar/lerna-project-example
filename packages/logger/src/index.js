import config from '@lerna-pro/config'

export const LOG_LEVEL = {
  NONE: -1,
  EVENT: 0, // system event
  CRASH: 1, // application failure
  ERROR: 2, // recoverable error
  WARNING: 3, // something is wrong
  INFO: 4, // overall explanation of what's happening
  DEBUG: 5 // for use in development
}

class Logger {
  constructor() {
    this._setLogLevel(LOG_LEVEL[config.logLevel] || LOG_LEVEL.NONE)
  }

  _setLogLevel(logLevel) {
    this._logLevel = logLevel
  }

  async _process(payload, logLevel) {
    if (logLevel <= this._logLevel) {
      console.log(payload)
    }
  }

  async crash(payload) {
    this._process(payload, LOG_LEVEL['CRASH'])
  }
  async error(payload) {
    this._process(payload, LOG_LEVEL['ERROR'])
  }
  async warning(payload) {
    this._process(payload, LOG_LEVEL['WARNING'])
  }
  async info(payload) {
    this._process(payload, LOG_LEVEL['INFO'])
  }
  async debug(payload) {
    this._process(payload, LOG_LEVEL['DEBUG'])
  }
  async event(payload) {
    this._process(payload, LOG_LEVEL['EVENT'])
  }
}

const logger = new Logger()

export default logger
