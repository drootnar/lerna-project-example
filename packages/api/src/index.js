import logger from '@lerna-pro/logger'
import config from '@lerna-pro/config'

const createApp = require('./app').createApp
const routes = require('./routes').default

const app = createApp(routes, true)
app.listen(config.api.port, function() {
  logger.info(`Api is listening on port ${config.api.port}`)
})
