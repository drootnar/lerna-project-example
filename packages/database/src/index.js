import redis from 'redis'
import bluebird from 'bluebird'

import config from '@lerna-pro/config'
import logger from '@lerna-pro/logger'

let _redis

export function getRedisClient () {
  if (_redis === undefined) {
    logger.debug('create redis client')
    _redis = redis.createClient(config.redisUrl)
    bluebird.promisifyAll(redis.RedisClient.prototype)
    bluebird.promisifyAll(redis.Multi.prototype)
  }
  return _redis
}

export function destroyRedisClient () {
  logger.debug('destroying redis client')
  if (_redis !== undefined) {
    _redis.end(true)
  }
  _redis = undefined
}

export default getRedisClient
