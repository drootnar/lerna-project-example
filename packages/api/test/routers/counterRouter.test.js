import request from 'supertest'
import { createApp } from '../../src/app'
import CounterRouter from '../../src/routers/counterRouter'

describe('CounterRouter', () => {
  const app = createApp(CounterRouter)

  it('POST v0/counter - checks response structure', async function () {
    const res = await request(app).post('/').send({ counter: 1 }).expect(200)
    expect(res.body).toMatchSnapshot()
  })

  it('GET v0/counter - checks response structure', async function () {
    const res = await request(app).get('/').expect(200)
    expect(res.body).toMatchSnapshot()
  })
})
