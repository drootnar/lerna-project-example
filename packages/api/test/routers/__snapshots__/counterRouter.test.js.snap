// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterRouter GET v0/counter - checks response structure 1`] = `
Object {
  "counter": "1",
}
`;

exports[`CounterRouter POST v0/counter - checks response structure 1`] = `
Object {
  "counter": "1",
}
`;
